src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Repository.kt

```diff
@@ -136,44 +136,41 @@ class Repository(
 
     val offender = offenderRepository.saveAndFlush(offenderBuilder.build(gender)).apply {
       rootOffenderId = id
+      rootOffender = this
     }
 
-    offender.bookings.addAll(
-      offenderBuilder.bookingBuilders.mapIndexed { index, bookingBuilder ->
-        val booking = bookingBuilder.build(offender, index, lookupAgency(bookingBuilder.agencyLocationId))
-        bookingBuilder.visitBalanceBuilder?.run {
-          booking.visitBalance = this.build(booking)
-        }
-        booking.contacts.addAll(
-          bookingBuilder.contacts.map {
-            it.build(booking, lookupContactType(it.contactTypeCode), lookupRelationshipType(it.relationshipTypeCode))
-          },
-        )
-        booking.visits.addAll(
-          bookingBuilder.visits.map { visitBuilder ->
-            val visit = visitBuilder.build(
-              offenderBooking = booking,
-              visitType = lookupVisitType(visitBuilder.visitTypeCode),
-              visitStatus = lookupVisitStatus(visitBuilder.visitStatusCode),
-              agencyLocation = lookupAgency(visitBuilder.agyLocId),
-              agencyInternalLocation = visitBuilder.agencyInternalLocationDescription?.run {
-                lookupAgencyInternalLocationByDescription(
-                  this,
-                )
-              },
-            )
-            visit.visitors.addAll(
-              visitBuilder.visitors.map {
-                it.build(it.person, leadVisitor = it.leadVisitor, visit)
-              } + visitBuilder.visitOutcome.build(visit),
-            )
-            visit
-          },
-        )
-
-        booking
-      },
-    )
```

Comment:
just removed the 'addAll' as the list is now added to inside the bookingBuilder.build()